review/fileToReview.js

```diff
@@ -12,62 +17,42 @@ var domain = "bank.local.fr"
  * @return {Object} All transactions available on the page
  */
 async function fetchTransactions(fromDate, authorization, jws = null, id, page, previousTransactions) {
-	console.log(`--- Fetch Trasactions page n°${page} ---`);
-	try {
-    var headers = {"Authorisation":  authorization }
+  console.log(`--- Fetch Trasactions page n°${page} ---`);
 
-    if (jws) {
-      headers = {
-        "Authorisation": authorization,
-        "jws": jws,
-        "Content-type": "application/json",
-        "Accept": "application/json"
-      }
-    } else {
-      headers = {
-        "Authorisation": authorization,
-        "Content-type": "application/json",
-        "Accept": "application/json",
-      }
-    }
+  const headers = jws ? {
+    ...config.defaultHeaders,
+    "jws": jws,
+    "Authorisation": authorization
+  } : {...config.defaultHeaders, "Authorisation": authorization};
+
+  const {code, response} = await doRequest('GET', `${config.domain}/accounts/${id}/transactions?page=${page}`, headers);
 
-	  var {code, response } = await doRequest('GET',
-      domain + '/accounts/'+ id + '/transactions?' + `page=${page}`,
-      headers);
+  if (code < 200 || code >= 400 || !response?.data) {
+    console.error(`Bad request received from bankin API : ${code}`, response);
+    return [];
+  }
+
+  if (response?.data?.meta && response?.data?.meta?.hasPageSuivante) {
+    const mouvements = response.data.Mouvements;
+    const date = mouvements[mouvements.length - 1].dateValeur;
+
+    if (date <= fromDate) {
+      console.log("FromDate is Reached - we don't need more transaction");
+    } else {
 
+      if (!mouvements)
+        throw new Error("Empty list of transactions ! " + JSON.stringify(previousTransactions));
 
-		if (response && code == 200 && response.data) {
-      if (response.data.meta) {
-        if (response.data.meta.hasPageSuivante) {
-          let mouvements = response.data.Mouvements;
-          var date = mouvements[mouvements.length -1].dateValeur;
-          if (date <= fromDate) {
-            console.log("FromDate is Reached - we don't need more transaction");
```

Comment:
I see a lot of console log and they also create a lot of extra IF / ELSE in your code. Are they all needed ? 
In case of debug i think this is useful but in case of production they are not and they are reducing the clarity of the code BUT also the clarity of the debugger.